src/init.c

```diff
@@ -378,6 +383,51 @@ shmem_internal_heap_postinit(void)
               shmem_internal_data_base, shmem_internal_data_length);
 
 #ifdef HAVE_SCHED_GETAFFINITY
+#ifdef USE_HWLOC
+    ret = hwloc_topology_init(&shmem_internal_topology);
+    if (ret < 0) {
```

Comment:
Should we also guard hwloc calls against `ret > 0`?

Reply:
I'm not sure that it's necessary - according to the hwloc documentation, the return value of the hwloc API calls being made will either be 0 (success) or -1 (error)

Reply:
Yeah I saw that... I suppose I prefer the tighter check because any ret > 0 is undefined hwloc behavior (which would pass these checks).  Maybe I'm being pedantic - in practice it should be fine.